### PR TITLE
Feature/cli module refactoring

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
Contains a big refactoring of the cli module. In specific:

- Minor refactoring all over the module based on Intellijs code analysis.
- A rewrite of the CollectedLogger class, because the base interface we used is deprecated since 2020.
- The CLI class has been split up into multiple classes.

### CODE REVIEW

File: cli/src/main/java/de/jplag/cli/picocli/CliInputHandler.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        this.options = new CliOptions();
+        this.commandLine = buildCommandLine();
+    }
+
+    private CommandLine buildCommandLine() {
+        CommandLine cli = new CommandLine(this.options);
+        cli.setHelpFactory(new HelpFactory());
+
+        cli.getHelpSectionMap().put(SECTION_KEY_OPTION_LIST, help -> help.optionList().lines().map(it -> {
+            if (it.startsWith("  -")) {
```
Comment (uuqjz) [inline cli/src/main/java/de/jplag/cli/picocli/CliInputHandler.java:65]: Can you replace these magic strings?

File: cli/src/main/java/de/jplag/cli/CLI.java [hunk 2]
```diff
@@ -1,290 +1,120 @@
...
-import picocli.CommandLine;
-import picocli.CommandLine.Model.CommandSpec;
-import picocli.CommandLine.Model.OptionSpec;
-import picocli.CommandLine.ParseResult;
 
 /**
  * Command line interface class, allows using via command line.
  * @see CLI#main(String[])
  */
 public final class CLI {
```
Comment (uuqjz) [inline cli/src/main/java/de/jplag/cli/CLI.java:24]: In all other classes you use `Cli` but here it is capitalized

**Review by uuqjz — APPROVED** [review]
Looks got to me.

**Review by tsaglam — COMMENTED** [review]
Have you tested that the CLI help text and the log output during the run is not changed?

File: cli/src/main/java/de/jplag/cli/CLI.java (hunk 2 above)
Comment (tsaglam) [reply cli/src/main/java/de/jplag/cli/CLI.java:24]: No, `CliFooBar` is the standard way to work with established abbreviations in Java identifiers.

**Review by tsaglam — APPROVED** [review]

**Pull request merged**